core/trino-parser/src/main/java/io/trino/sql/tree/PlanSiblings.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class PlanSiblings
```

Comment:
Instead of two classes (PlanParentChild and PlanSiblings) that are further parameterized with an enum, I might opt to go with four classes that represent the specific plan types: OuterJsonPlan, InnerJsonPlan, CrossJsonPlan, UnionJsonPlan. But I need to see how this fits in the query analysis and planning steps.

Reply:
I don't think we should do it. It would complicate the analysis and planning. The analyzer does not look at the types (enums) at all. Planning is also more convenient with less classes.